functions/src/models/requests/timeline.ts

```diff
@@ -0,0 +1,96 @@
+import { FirestoreDataConverter } from '@google-cloud/firestore';
+import { IsEnum, IsObject } from 'class-validator';
+import { firestore } from 'firebase-admin';
+
+import { IOffer } from '../offers';
+import { IRequest, RequestStatus } from './index';
+import Timestamp = firestore.Timestamp;
+import DocumentData = firestore.DocumentData;
+
+export interface ITimelineItem extends DocumentData {
+  offerSnapshot: IOffer;
+  requestSnapshot: IRequest;
+  status: RequestStatus;
```

Comment:
`status` is not necessary if it's included in `requestSnapshot` right? which will include even more information?

Unless `requestSnapshot` is the (previous) snapshot? But i think it makes more sense to have it be the *new* state of the request...

We would need to make sure that we create a timeline item for the initial creation of requests in that case (should be pretty easy with an event)

```suggestion
```